presto-main/src/main/java/io/prestosql/operator/GroupedTopNBuilder.java

```diff
@@ -78,17 +53,20 @@ public GroupedTopNBuilder(
     {
         this.sourceTypes = requireNonNull(sourceTypes, "sourceTypes is null");
         checkArgument(topN > 0, "topN must be > 0");
-        this.topN = topN;
         this.produceRowNumber = produceRowNumber;
         this.groupByHash = requireNonNull(groupByHash, "groupByHash is not null");
 
         requireNonNull(comparator, "comparator is null");
-        this.comparator = (left, right) -> comparator.compareTo(
-                pageReferences.get(left.getPageId()).getPage(),
-                left.getPosition(),
-                pageReferences.get(right.getPageId()).getPage(),
-                right.getPosition());
-        this.emptyPageReferenceSlots = new IntFIFOQueue();
+        groupedTopNRowNumberAccumulator = new GroupedTopNRowNumberAccumulator(
+                (rowId1, rowId2) -> {
+                    Page page1 = pageManager.getPage(rowId1);
```

Comment:
Normally we name stuff like "leftPage" and "righttPage". The actual left and right just argument order but it doesn't really matter, but it make it easier to see than a single number.  The is especially true when the argument names get longer.